app/models/concerns/course/lesson_plan_concern.rb

```diff
@@ -0,0 +1,16 @@
+module Course::LessonPlanConcern
+  extend ActiveSupport::Concern
+
+  # Before the lesson plan is displayed, lesson plan items are ordered by start
+  # date and divided into groups separated by milestones. Each group, possibly
+  # except the first, is an array that has a milestone as its first item.
+  # This method returns such a list of groups for a course.
+  # @return [[Array<Array<Course::LessonPlanMilestone, Course::LessonPlanItem>>]]
```

Comment:
I think your type declaration is wrong... this is an array of an array of an array of either milestones or items.